api/src/OpenTelemetry/Internal/Logs/Types.hs

```diff
@@ -19,18 +20,92 @@ module OpenTelemetry.Internal.Logs.Types (
 
 import Control.Concurrent.Async
 import Data.Function (on)
+import Data.HashMap.Strict (HashMap)
 import qualified Data.HashMap.Strict as H
 import Data.IORef (IORef, atomicModifyIORef, modifyIORef, newIORef, readIORef)
 import Data.Text (Text)
 import Data.Vector (Vector)
 import OpenTelemetry.Common (Timestamp, TraceFlags)
 import OpenTelemetry.Context.Types (Context)
-import OpenTelemetry.Internal.Common.Types (InstrumentationLibrary, ShutdownResult)
+import OpenTelemetry.Internal.Common.Types (ExportResult, InstrumentationLibrary, ShutdownResult)
 import OpenTelemetry.Internal.Trace.Id (SpanId, TraceId)
 import OpenTelemetry.LogAttributes
 import OpenTelemetry.Resource (MaterializedResources)
 
 
+{- | LogRecordExporter defines the interface that protocol-specific exporters must implement so that they can be plugged into OpenTelemetry SDK and support sending of telemetry data.
+
+The goal of the interface is to minimize burden of implementation for protocol-dependent telemetry exporters. The protocol exporter is expected to be primarily a simple telemetry data encoder and transmitter.
+-}
+data LogRecordExporter body = LogRecordExporter
+  { logRecordExporterExport :: HashMap InstrumentationLibrary (Vector (LogRecord body)) -> IO ExportResult -- TODO: This should be ReadableLogRecord instead of LogRecord
```

Comment:
[This must be updated](https://opentelemetry.io/docs/specs/otel/logs/sdk/#export) to be a `ReadableLogRecord` once that type exists in a suitable format for this use case.